relay-profiling/src/android.rs

```diff
@@ -57,6 +60,29 @@ impl AndroidProfile {
         };
         Ok(())
     }
+
+    /// Removes an event with a duration of 0
+    fn remove_events_with_no_duration(&mut self) {
+        let mut android_profile = self.profile.as_ref().unwrap().clone();
```

Comment:
Do we really need to clone here?
```suggestion
        let android_profile = self.profile.as_mut().unwrap();
```

Also, the `unwrap()` is unfortunate, even if calling `parse()` guarantees that the option has a value. Maybe we can make `AndroidProfile::profile` non-optional, and convert `parse()` into a function that is called during deserialization, something like

```rust
#[serde(deserialize_with="parse_android_trace_log")]
profile: AndroidTraceLog,
```

Reply:
It is unfortunate and the reason for that was to not have to parse if the `relay` node was not in `processing` mode. Now, I think we can still make it so we only parse when we need to while still getting rid of the `Option`.